File: Day13/solution.ts
function reverseVowels(s: string): string {
    let pos = []; let arrayS = s.split('');
    const swap = (first: number,last:number) => {
    if(arrayS[first]===arrayS[last]) return
    let temp = arrayS[first]; arrayS[first] = arrayS[last]; arrayS[last] = temp;}
    for (let i = 0; i< s.length; i++) {
    if (['a','e','i','o','u','A','E','I','O','U'].includes(s )) pos.push(i)}

File: Day13/solution.cpp
class Solution {
public:
    string reverseVowels(string s) {
        unordered_set<char> vowels = {'a','e','i','o','u','A','E','I','O','U'};
        int i = 0, j = s.size() - 1;
        while(i < j){
            while(i < j && !vowels.count(s[i])) i++;
            while(i < j && !vowels.count(s[j])) j--;
            if(i < j) swap(s[i++], s[j--]);
        }
        return s;
    }
};